### PR TITLE
Refactor x509 extension embedding logic

### PR DESCRIPTION
**Pull request by nsmith5**
#### Summary

Adds a new extensions structure that documents the global set of
extensions added to our certificates and a method to render that data
into extensions.

#### Ticket Link

Related to #275 

#### Release Note

```release-note
* NONE
```

### CODE REVIEW

File: pkg/ca/x509ca/common_test.go [hunk 1]
```diff
@@ -36,3 +47,102 @@ func TestGenerateSerialNumber(t *testing.T) {
...
+	return priv.Public()
+}
+
+func TestMakeCert(t *testing.T) {
+	tests := map[string]struct {
+		Challenge challenges.ChallengeResult
+		WantErr   bool
+		WantFacts map[string]func(x509.Certificate) error
+	}{
+		`Github workflow challenge should have all Github workflow extensions set`: {
```
Comment (haydentherapper) [inline pkg/ca/x509ca/common_test.go:65]: nit: Worth checking that issuer is always set in the GitHub workflow too?

File: pkg/ca/x509ca/extensions.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+// Extensions contains all custom x509 extensions defined by Fulcio
+type Extensions struct {
+	// NB: New extensions must be added here and documented
+	// at docs/oidc-info.md
+
+	// The OIDC issuer. Should match `iss` claim of ID token or, in the case
+	// of a federated login like Dex it should match the issuer URL of the upstream
+	// issuer
+	Issuer string // OID 1.3.6.1.4.1.57264.1.1
```
Comment (haydentherapper) [inline pkg/ca/x509ca/extensions.go:31]: Can you add a comment that `Issuer` must be set?

**Review by haydentherapper — APPROVED** [review]
This looks great!

File: pkg/ca/x509ca/common_test.go (hunk 1 above)
Comment (nsmith5) [reply pkg/ca/x509ca/common_test.go:65]: Yup the test is there I just didn't add that to the description.

```
`Certifificate should have correct issuer`:     factIssuerIs(`https://token.actions.githubusercontent.com`),
````

**Review by haydentherapper — APPROVED** [review]

**Pull request merged**